rcl_action/src/rcl_action/action_client.c

```diff
@@ -148,9 +200,12 @@ rcl_action_client_init(
   RCL_CHECK_FOR_NULL_WITH_MSG(
     action_client->impl, "allocating memory failed", return RCL_RET_BAD_ALLOC);
 
+  // Avoid uninitialized pointers should initialization fail
+  *action_client->impl = _rcl_action_get_zero_initialized_client_impl();
   // Copy action client name and options.
   action_client->impl->action_name = rcutils_strdup(action_name, allocator);
   if (NULL == action_client->impl->action_name) {
+    RCL_SET_ERROR_MSG("failed to duplicate action name");
```

Comment:
This might be overwritten by https://github.com/ros2/rcl/pull/717/files?file-filters%5B%5D=.c#diff-1f30210b7875b9b39020ed2b9c12bd78L173

Reply:
Are there any specific circumstances you are worried about? Setting impl to get the zero_initialized values should put the object into a known state, which helps ensure cleanup occurs predictably. Otherwise if there is a double error, then it can be helpful for the message to be set twice, so that the log file has a record of the sequence of events that failed.

Reply:
Actually no, but there's been discussion about this, https://github.com/ros2/rcl/issues/713#issuecomment-657746623. I do not think that this is the problem but status. I'd think it might be better to share.